src/App.jsx

```diff
@@ -14,70 +16,75 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedProduct, setProduct] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const isSelectedProduct = product => product === selectedProduct;
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  return (
+    <main className="section container">
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      {
+        selectedProduct
+          ? (
+            <h1 className="title is-flex is-align-items-center">
+              {`${selectedProduct} is selected`}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => setProduct('')}
+              />
+            </h1>
+          )
+          : (
+            <h1 className="title is-flex is-align-items-center">
+              No goods selected
+            </h1>
+          )
+      }
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+      <table className="table">
+        <tbody>
+          {goods.map(product => (
+            <tr
+              data-cy="Good"
+              key={product}
+              className={
+                cn({
+                  'has-background-success-light':
+                    isSelectedProduct(product),
+                })
+              }
             >
-              -
-            </button>
-          </td>
+              <td>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+                <button
+                  data-cy={isSelectedProduct(product)
+                    ? 'RemoveButton'
+                    : 'AddButton'}
+                  type="button"
+                  className={cn('button',
+                    { 'is-info': isSelectedProduct(product) })}
+                  onClick={() => {
+                    if (isSelectedProduct(product)) {
```

Comment:
Move this logic to the separate function 
Use ternary operator instead of if else as it will make a code shorter